src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -156,13 +207,22 @@ protected void containerIsStarting(final InspectContainerResponse containerInfo,
 
         StringBuilder kafkaConfiguration = new StringBuilder();
 
-        // default listener config
-        kafkaConfiguration
-            .append(" --override listeners=").append(kafkaListeners).append("PLAINTEXT://0.0.0.0:").append(KAFKA_PORT)
-            .append(" --override advertised.listeners=").append(advertisedListeners)
-            .append(" --override zookeeper.connect=localhost:").append(ZOOKEEPER_PORT)
-            .append(" --override listener.security.protocol.map=").append(kafkaListenerSecurityProtocol).append("PLAINTEXT:PLAINTEXT")
-            .append(" --override inter.broker.listener.name=BROKER1");
+        if (!useKraft) {
+            // default listener config
+            kafkaConfiguration
+                    .append(" --override listeners=").append(kafkaListeners).append("PLAINTEXT://0.0.0.0:").append(KAFKA_PORT)
+                    .append(" --override advertised.listeners=").append(advertisedListeners)
+                    .append(" --override zookeeper.connect=localhost:").append(ZOOKEEPER_PORT)
```

Comment:
As I can see the only thing that differs from using KRaft is `.append(" --override zookeeper.connect=localhost:").append(ZOOKEEPER_PORT)`. Would It be better to make 
```
kafkaConfiguration
                    .append(" --override listeners=").append(kafkaListeners).append("PLAINTEXT://0.0.0.0:").append(KAFKA_PORT)
                    .append(" --override advertised.listeners=").append(advertisedListeners)
                    .append(" --override listener.security.protocol.map=").append(kafkaListenerSecurityProtocol).append("PLAINTEXT:PLAINTEXT")
                    .append(" --override controller.listener.names=").append("BROKER1")
                    .append(" --override inter.broker.listener.name=BROKER1");
```
and afterwards use decision logic to only append `.append(" --override zookeeper.connect=localhost:").append(ZOOKEEPER_PORT)` if we don't use Kraft?

Reply:
There's also `.append(" --override controller.listener.names=").append("BROKER1")` in the Kraft case, but I agree we should try to keep the common configs out of the `if/else`.

Reply:
Fixed.